src/Controllers/BatchProcessorController.ts

```diff
@@ -475,6 +480,86 @@ module StreamStats.Controllers {
+            this.submittingBatch = true;
+            
+            // handles submitting more than 250 points
+            if (submit250 == true) {
+
+                // appends moreThan250Points to formdata
+                formdata.append('moreThan250Points', submit250.toString());
+
+                // submits batch to API
+                this.postBatchFormData(formdata, headers).then(
+                    response => {
+                        var r = response;
+                        if (r.status == 200) {
+                            this.toaster.pop('success', "The batch was submitted successfully. You will be notified by email when results are available.", "", 20000);
+                        }
+
+                        // handle if status is not 200, let user know error
+                        else {
+                            var detail = r.data.detail
+                            this.toaster.pop('error', "The submission failed for the following reason:" + detail, "", 5000);
+                        }
+                    }).finally(() => {
+                    
+                    this.submittingBatch = false;
+                    this.submitBatchOver250 = false;
+                });
+            }
+
+            else { 
+                
+                // submits batch to API
+                this.postBatchFormData(formdata, headers).then(
+                    response => {
+                        // assign response to variable
+                        var r = response;
+
+                        // first, handle if status is 500 and detail contains more "250", which means use tried to submit more than 250 points
+                        if (r.status == 500 && r.data.detail.indexOf("250") > -1) {
+                            let message = "Batch contains more than 250 points. Only the first 250 points will be processed. Please resubmit the batch if you would like only the first 250 points to be processed."
+                            this.submitBatchOver250 = true;
+                            this.toaster.pop('warning', message, "", 5000);
+                        }
+                        
+                        // handle if status is 200, which means the batch was submitted successfully
+                        else if (r.status == 200) {
+                            this.toaster.pop('success', "The batch was submitted successfully. You will be notified by email when results are available.", "", 20000);
+                        }
+                        
+                        // handle if status is not 200 or to do with 250 points and let submitter know error
+                        else {
+                            var detail = r.data.detail
+                            this.toaster.pop('error', "The submission failed for the following reason:" + detail, "", 5000);
+                        }
+
+                    }).finally(() => { this.submittingBatch = false; });

```

Comment:
Also refresh the Batch Status tab if an email address has already been entered. How to recreate the "error" (not really an error but may be confusing for users):

1. Go to Batch Status tab
2. Enter email and look at my batches
3. Go to Submit Batch tab
4. Submit a batch
5. Go back to Batch Status tab
6. My new batch isn't showing up yet; I have to click the "Check Status" button again to refresh the list